### DISCUSSION TITLE
Missing hadm_id and lack of lab draws in labevents table for large portion of dataset?

### DISCUSSION THREAD

**Discussion by mxa256** [MIT-LCP/mimic-code#1758] [MIMIC-IV] [answered]
Hello, 
I have a cohort created with MIMIC-IV (n~5500) however I have two problems. 
1. The hadm_id is missing from a large number of observations. Per the documentation, I did impute using the admissions table, but this did not really help, as the number of imputed values was very small (about 0.5% of observations). 
2. After imputing and joining, I filtered for lab observations that were drawn between the admissions admittime and dischtime. But as a result there are lab observations for only about ~130 patients. Has anyone else encountered issues with sparse lab observations? 
Happy to share my code if needed.

Mona

**Comment by shujuecn** [comment]
Hi there,

I recently upgraded to MIMIC IV version 3.1 and came across the issue mentioned in your title. 
Specifically, I ran the following query:

```sql
SELECT 
    COUNT(*) AS total_rows,
    COUNT(hadm_id) AS non_null_rows,
    COUNT(*) - COUNT(hadm_id) AS null_rows
FROM 
    mimiciv_hosp.labevents;
```

The results were:

| total_rows | non_null_rows | null_rows |
|------------|---------------|-----------|
| 158374764  | 84605867      | 73768897  |

I would like to know if this result matches yours. I'm even starting to question whether there was an issue with my database installation. 
Additionally, have you done any further explorations to address this issue? I look forward to your response.

**Comment by marymlucas** [comment]
Hi @shujuecn 
I can confirm it's not an issue with your database installation. I am able to reproduce your counts using the labevents CSV file for MIMIC-IV version 3.1 from Physionet. I used Python and pandas:

![image](https://github.com/user-attachments/assets/9654d262-a53b-484d-8de7-3384f9518669)

**Reply by shujuecn** [reply]
Thank you for your confirmation; it was very helpful to me.

**Comment by marymlucas** [comment, marked as answer]
I believe this is related to this closed issue #1823 which @alistairewj responded to with an explanation:

![image](https://github.com/user-attachments/assets/727efe82-6515-4666-97f1-0d8c3e069ce2)

**Comment by mxa256** [comment]
Hello, I filtered the dataset to construct a cohort based on hadmids before I examined the lab events table. The cohort only included patients with a hospitalization, so issue 1823 does not apply. I think it’s just poor data quality. I had to exclude patients with missing values which really dwindled down my cohort.

**Reply by marymlucas** [reply]
Only other thing I can think of is dates not being concordant, i.e. inpatient time vs labevents time.  But I'm just grasping at straws here. Do you have your code somewhere? I could take a stab at it. 

For what it's worth I reran the query using the version of MIMIC-IV currently on BigQuery, which I think is version 2.x, and the proportion of missing hadm_ids in the labevents table in that version seems similar to the ones in v3, ~ 47%  So perhaps it is a data quality issue as you suspect, or we're just missing something.

![image](https://github.com/user-attachments/assets/713fe1f1-c159-4649-b82d-4b7d551d00f4)

**Reply by mxa256** [reply]
Hi Mary, yes when I coded, I made sure the draw date for the lab was in between the hospital admission date and discharge date for the associated hadm_id. I will share some code later.
